apps/ai-content-generator/src/locations/ConfigScreen.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import { Form, FormControl, TextInput, Textarea, Flex, Select } from '@contentful/f36-components';
+import { css } from 'emotion';
+import { useSDK } from '@contentful/react-apps-toolkit';
+
+const ConfigScreen = () => {
+  const [parameters, setParameters] = useState({});
+  const [apiKey, setApiKey] = useState('');
+  const [editing, setEditing] = useState(false);
+  const sdk = useSDK();
+  const onConfigure = useCallback(async () => {
+    const currentState = await sdk.app.getCurrentState();
+    setApiKey('');
+    setEditing(false);
+    return {
+      parameters,
+      targetState: currentState,
+    };
+  }, [parameters, sdk]);
+
+  useEffect(() => {
+    sdk.app.onConfigure(() => onConfigure());
+  }, [sdk, onConfigure]);
+
+  useEffect(() => {
+    (async () => {
+      const currentParameters = await sdk.app.getParameters();
+      if (currentParameters) {
+        setParameters(currentParameters);
+      }
+      sdk.app.setReady();
+    })();
+  }, [sdk]);
```

Comment:
React in StrictMode might yell about this, because it could introduce a memory leak if the component is remounted (I'm not sure if it will though). This is how to handle asynchronous calls in a useEffect if StrictMode gives you any issues:
```suggestion
  useEffect(() => {
    let isMounted = true;
    
    (async () => {
      const currentParameters = await sdk.app.getParameters();
      if (currentParameters) {
        setParameters(currentParameters);
      }
      sdk.app.setReady();
    })();
    
    return () => {
      isMounted = false;
    }
  }, [sdk]);
```

Reply:
done